doc/source/skips/4-transition-to-v2.rst

```diff
@@ -0,0 +1,249 @@
+- Harmonizing similar parameters in different functions to have the same name;
+  for example, we currently have ``random_seed``, ``random_state``, ``seed``,
+  or ``sample_seed`` in different functions, all to mean the same thing.
+- Changing ``measure.regionprops`` to return a dictionary instead of a list.
+- Combine functions that have the same purpose, such as ``watershed``,
+  ``slic``, or ``felzenschwalb``, into a common namespace. This would make it
+  easier for new users to find out which functions they should try out for a
+  specific task. It would also help the community grow around common APIs,
+  where now scikit-image APIs are essentially unique for each function.
+
+To make this transition with a minimum amount of user disruption, this SKIP
+proposes releasing a new library, skimage2, that would replace the existing
+library, *but only if users explicitly opt-in*. Additionally, by releasing a
+new library, users could depend *both* on scikit-image (1.0) and on skimage2,
+allowing users to migrate their code progressively.
+
+Related Work
+------------
+
+`pandas` released 1.0.0 in January 2020, including many backwards-incompatible
+API changes [3]_. `scipy` released version 1.0 in 2017, but, given its stage of
+maturity and position at the base of the scientific Python ecosystem, opted not
+to make major breaking changes [4]_. However, SciPy has adopted a policy of
+adding upper-bounds on dependencies [5]_, acknowledging that the ecosystem as a
+whole makes backwards incompatible changes on a 2 version deprecation cycle.
+
+Several libraries have successfully migrated their user community to a new
+namespace with a version number on it, such as OpenCV (imported as ``cv2``) and
+BeautifulSoup (imported as ``bs4``), Jinja (``jinja2``) and psycopg (currently
+imported as ``psycopg2``). Further afield, R's ggplot is used as ``ggplot2``.
+
+Implementation
+--------------
+
+The details of the proposal are as follows:
+
+- scikit-image 0.19 will be followed by scikit-image 1.0. Every deprecation
+  message will be removed from 1.0, and the API will be considered the
+  scikit-image 1.0 API.
+- After 1.0, the main branch will be changed to (a) change the import name to
+  skimage2, (b) change the package name to skimage2, and (c) change the version
+  number to 2.0-dev.
+- There will be *no* scikit-image package on PyPI with version 2.0. Users who
+  ``pip install scikit-image`` will always get the 1.0 version of the package.
+- After consensus has been reached on the new API, skimage2 will be released.
+- scikit-image 1.0.x will receive critical bug fixes for an unspecified period
+  of time, depending on the severity of the bug and the amount of effort
+  involved.
+
+Backward compatibility
+----------------------
+
+This proposal breaks backward compatibility in numerous places in the library.
+However, it does so in a new namespace, so that this proposal does not raise
+backward compatibilty concerns for our users.
```

Comment:
I propose adding some text to indicate we will try to not introduce more incompatible changes than necessary and will provide a transition guide.

```suggestion
This proposal breaks backward compatibility in numerous places in the library.
However, it does so in a new namespace, so that this proposal does not raise
backward compatibilty concerns for our users. That said, the authors will
attempt to limit the number of backward incompatible changes to those likely to
substantially improve the overall user experience. It is anticipated that
porting `skimage` code to `skimage2` will be a straightforward process
and we will publish a user guide for making the transition by the time of
the `skimage2` release.
```